### PR TITLE
DOCS: Update function body comment of pattern matcher

### PR DESCRIPTION
**Pull request by sjokkateer**

### CODE REVIEW

**Comment by sjokkateer** [comment]
Please forgive me but I'm having a hard time referencing the exact line of the docs through the web interface.

Trying the pattern matching example `$router->addPatternMatcher('wordStartsWithM', '(m|M)[a-zA-Z]+');` out for myself, I got the following exception 
> Fatal error: Uncaught FastRoute\BadRouteException: Regex "(m|M)[a-zA-Z]+" for parameter "name"

The FastRoute documentation mentions the following
> Custom patterns for route placeholders cannot use capturing groups. For example {lang:(en|de)} is not a valid placeholder, because () is a capturing group. Instead you can use either {lang:en|de} or {lang:(?:en|de)}.

Would you like me to update the example to use a smiliar pattern `(?:m|M)[a-zA-Z]+` or am I missing out on something?

If I overlooked something, or if there are any additions you would like me to make within same scope please let me know!

**Comment by philipobenito** [comment]
Hi @sjokkateer there may have been an update in FastRoute that we missed or the example wasn't fully tested when the PR was initially sent.

I'm happy for you to update the docs, could you please also add a unit test for whichever example you decide on?

**Comment by sjokkateer** [comment]
Hi @philipobenito are these tests the tests according to your expectations? If not, could you point me into the right direction.

If yes, should I also add some counter examples when I got time?

File: docs/5.x/routes.md [hunk 1]
```diff
@@ -222,19 +222,18 @@ use Psr\Http\Message\ServerRequestInterface;
 
 $router = new League\Route\Router;
 
-$router->addPatternMatcher('wordStartsWithM', '(m|M)[a-zA-Z]+');
+$router->addPatternMatcher('wordStartsWithM', '(?:m|M)[a-zA-Z]+');
```
Comment (philipobenito) [inline docs/5.x/routes.md:225]: nicely done with the non-capturing group

**Review by philipobenito — APPROVED** [review]

**Comment by philipobenito** [comment]
@sjokkateer this looks great thanks, I may switch it to using the PHPUnit API to set the data provider rather than annotation, but other than that looks good 👍 

More than happy for you to build out more examples and counter examples.

**Pull request merged**